lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb

```diff
@@ -1257,26 +1250,33 @@ fc_extras
         if isinstance(cf_coord_var, cf.CFLabelVariable):
             points_data = cf_coord_var.cf_label_data(cf_var)
         else:
-            # Use a closure to capture the current file/variable names.
-            filename = engine.filename
-            nc_var_name = cf_coord_var.cf_name
-            get_values = lambda: deferred_load(filename, nc_var_name)
-            points_data = iris.aux_factory.LazyArray(cf_coord_var.shape,
-                                                     get_values)
+            dtype = cf_coord_var.dtype
+            fill_value = getattr(cf_coord_var.cf_data, '_FillValue',
+                                 netCDF4.default_fillvals[dtype.str[1:]])
+            proxy = iris.fileformats.netcdf.NetCDFDataProxy(
+                cf_coord_var.shape, dtype, engine.filename,
+                cf_coord_var.cf_name, fill_value)
+            points_data = biggus.OrthoArrayAdapter(proxy)
 
         # Get any coordinate bounds.
         cf_bounds_var = get_cf_bounds_var(cf_coord_var)
         if cf_bounds_var is not None:
-            # Use a closure to capture the current file/variable names.
-            filename = engine.filename
-            nc_bounds_var_name = cf_bounds_var.cf_name
-            get_values = lambda: deferred_load(filename, nc_bounds_var_name)
-            bounds_data = iris.aux_factory.LazyArray(cf_bounds_var.shape,
-                                                     get_values)
+            dtype = cf_bounds_var.dtype
+            fill_value = getattr(cf_bounds_var.cf_data, '_FillValue',
+                                 netCDF4.default_fillvals[dtype.str[1:]])
+            proxy = iris.fileformats.netcdf.NetCDFDataProxy(
+                cf_bounds_var.shape, dtype, engine.filename,
+                cf_bounds_var.cf_name, fill_value)
+            bounds_data = biggus.OrthoArrayAdapter(proxy)
```

Comment:
@rhattersley is it worthwhile having a private convenience function to generate the `biggus.OrthoArrayAdapter` for the `cf_coord_var` or the `cf_bounds_var` ... something along the lines of;

```python
from iris.fileformats.netcdf import NetCDFDataProxy as Proxy
...
def _to_biggus(engine, cf_var):
    dtype = cf_var.dtype
    fill_value = getattr(cf_var.cf_data, '_FillValue', netCDF4.default_fillvals[dtype.str[1:]])
    proxy = Proxy(cf_var.shape, dtype, engine.filename, cf_var.cf_name, fill_value)
    return biggus.OrthoArrayAdapter(proxy)
```